.github/workflows/ci.yml

```diff
@@ -97,7 +106,8 @@ jobs:
     runs-on:
       # Pseudo-ternary hack and order matters. See comment at top of file.
       - self-hosted
-      - runner-group=${{ github.event_name == 'pull_request' && 'presubmit' || 'postsubmit' }}
+      - runner-group=${{ needs.should_run.outputs.runner-group }}
+      - environment=${{ needs.should_run.outputs.environment }}
```

Comment:
A bunch of these comments can be removed now, right? (the ternary hack for `runner-group` is precomputed in `should_run`)

If there is just one place with the hack, you could move the comment inline, rather than point "at top of file"

Reply:
> A bunch of these comments can be removed now, right? (the ternary hack for `runner-group` is precomputed in `should_run`)

Good point, fixed
 
> If there is just one place with the hack, you could move the comment inline, rather than point "at top of file"

There are multiple places with the ternary hack.

Reply:
I've actually pulled the other place we use the pseudo-ternary into the outputs here. I still think it maybe makes sense to comment on it in general since it's basically a "language feature" that we may use elsewhere as well.

Reply:
Oh wait, we still use it for GCS bucket too. We could move that one as well, but its location in env already makes it usable everywhere we want to use it. Actually, that's true of `write-caches` also, so I should either put that one in env also or put everything in `should_run` outputs. Referencing the latter is way more verbose, which is one reason not to

Reply:
Wait, but I also wanted your opinion on this question @ScottTodd 😁

Reply:
Oh, whoops :P thanks for the ping.

I think "should_run" could be renamed if it's taking on those other responsibilities, like "startup", "setup", or "prepare_env", etc.

Maybe we could pick a shorter name if referencing ends up being really verbose

Reply:
That's fair. Do you think we should stick all "global-ish" variables here, or put the ones that can go in env in env? The issue isn't really the name of the job, it's that instead of `env.VAR` we have `needs.should_run.outputs.VAR`

Reply:
`env.VAR` is nice and easy to type/read, and environment variables are portable to other CI systems...

consistency is also nice though

Reply:
Actually, it's funny because "env" can be used in "outputs", so for something we want to share between outputs, the cleanest thing to do is to put them in env :-/

Reply:
Chatted offline